act/registry_test.go

```diff
@@ -456,11 +457,136 @@ func Test_Run_ActionNotExist(t *testing.T) {
 	logger := zerolog.New(&buf)
 	actRegistry := NewActRegistry(
 		BuiltinSignals(), BuiltinPolicies(), BuiltinActions(),
-		config.DefaultPolicy, config.DefaultPolicyTimeout, logger)
+		config.DefaultPolicy, config.DefaultPolicyTimeout, config.DefaultActionTimeout, logger)
 	assert.NotNil(t, actRegistry)
 
 	result, err := actRegistry.Run(&sdkAct.Output{}, WithLogger(logger))
 	assert.Nil(t, result)
 	assert.Equal(t, err, gerr.ErrActionNotExist)
 	assert.Contains(t, buf.String(), "Action does not exist, run aborted")
 }
+
+// Test_Run_Timeout tests the Run function of the action with a timeout.
+func Test_Run_Timeout(t *testing.T) {
+	tests := []struct {
+		name           string
+		isAsync        bool
+		expectedErr    error
+		expectedLog    string
+		expectedResult any
+		timeout        time.Duration
+	}{
+		{
+			name:           "sync action",
+			isAsync:        false,
+			expectedErr:    gerr.ErrRunningActionTimeout,
+			expectedLog:    "",
+			timeout:        1 * time.Millisecond,
+			expectedResult: nil,
+		},
+		{
+			name:           "async action",
+			isAsync:        true,
+			expectedErr:    gerr.ErrAsyncAction,
+			expectedLog:    "{\"level\":\"error\",\"action\":\"waitAsync\",\"message\":\"Action timed out\"}",
+			timeout:        1 * time.Millisecond,
+			expectedResult: nil,
+		},
+		{
+			name:           "action with no timeout",
+			isAsync:        false,
+			expectedErr:    nil,
+			expectedLog:    "",
+			timeout:        0,
+			expectedResult: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			isAsync := test.isAsync
+			name, actions, signals, policies := createWaitActEntities(isAsync)
+			out := bytes.Buffer{}
+			logger := zerolog.New(&out)
+			actRegistry := NewActRegistry(
+				signals, policies, actions,
+				config.DefaultPolicy, config.DefaultPolicyTimeout, test.timeout, logger)
+			assert.NotNil(t, actRegistry)
+
+			outputs := actRegistry.Apply([]sdkAct.Signal{*signals[name]})
+			assert.NotNil(t, outputs)
+			assert.Equal(t, name, outputs[0].MatchedPolicy)
+			assert.Equal(t,
+				map[string]interface{}{
+					"async":   isAsync,
+					"level":   "info",
+					"log":     true,
+					"message": "test",
+				},
+				outputs[0].Metadata,
+			)
+			assert.Equal(t, !isAsync, outputs[0].Sync)
+			assert.True(t, cast.ToBool(outputs[0].Verdict))
+			assert.False(t, outputs[0].Terminal)
+
+			result, err := actRegistry.Run(outputs[0], WithResult(map[string]any{}))
+			if test.expectedErr != nil {
+				require.NotNil(t, err)
+				assert.ErrorIs(t, err, test.expectedErr)
+			} else {
+				require.Nil(t, err)
+			}
+			assert.Equal(t, test.expectedResult, result)
+
+			if isAsync {
+				time.Sleep(3 * time.Millisecond)
+			}
+			if test.expectedLog != "" {
+				assert.Contains(t, out.String(), test.expectedLog)
+			}
+		})
+	}
+}
+
+func createWaitActEntities(async bool) (
```

Comment:
Nit: move this function to a separate file: `act/act_helpers_test.go`.

Reply:
moved it to the file as you said